src/test/decompress_test.cpp

```diff
@@ -110,29 +110,18 @@ auto main(int, char* argv[]) -> int
         'd');
     std::span<const std::byte> src{compressed};
 
-    constexpr auto expected_0 = huffman::byte_array('r', 'o', 's', 'e');
-    constexpr auto expected_1 = huffman::byte_array('b', 'u', 'd');
-    const std::array<std::span<const std::byte>, 2> expecteds{
-        expected_0, expected_1};
-
-    std::array<std::byte, 4> dst_array{};
-    const std::span<std::byte> dst{dst_array};
-    for (std::size_t i = 0; i < expecteds.size(); ++i) {
-      const auto result = decompress(src, dst);
-      expect(result.has_value())
-          << "got error code: " << static_cast<std::int32_t>(result.error());
-      if (i == 0) {
-        expect(not result->remaining_src.empty());
-        expect(result->min_next_dst_size == expecteds.at(1).size());
-      } else {
-        expect(result->remaining_src.empty());
-        expect(result->min_next_dst_size == 0);
-      }
-      const auto expected = expecteds.at(i);
-      expect(result->dst_written == expected.size());
-      expect(std::ranges::equal(dst.subspan(0, expected.size()), expected));
-      src = result->remaining_src;
-    }
+    constexpr auto expected =
+        huffman::byte_array('r', 'o', 's', 'e', 'b', 'u', 'd');
+
+    std::array<std::byte, expected.size()> dst_array{};
+    std::span<std::byte> dst_too_small{dst_array.data(), dst_array.size() - 1};
+    const auto status_too_small = decompress(src, dst_too_small);
+    expect(status_too_small == DecompressStatus::DstTooSmall);
+
+    std::span<std::byte> dst{dst_array};
+    const auto status = decompress(src, dst);
+    expect(status == DecompressStatus::Success);
+    expect(std::ranges::equal(dst, expected));
```

Comment:
I think you can use:
```suggestion
    expect(range_eq(dst, expected));
```

https://github.com/garymm/starflate/blob/51a2ddbab9ae6f28de98061045199a9142588c4d/ut.hpp#L147